v4/feature/trade/src/main/java/exchange/dydx/trading/feature/trade/margin/DydxAdjustMarginInputViewModel.kt

```diff
@@ -23,59 +26,60 @@ class DydxAdjustMarginInputViewModel @Inject constructor(
     private val localizer: LocalizerProtocol,
     private val abacusStateManager: AbacusStateManagerProtocol,
     private val formatter: DydxFormatter,
-    private val router: DydxRouter,
     savedStateHandle: SavedStateHandle,
     private val parser: ParserProtocol,
 ) : ViewModel(), DydxViewModel {
 
-    private val marketId: String? = savedStateHandle["marketId"]
+    private val marketId: String = requireNotNull(savedStateHandle["marketId"]) { "Navigated to Adjust Margin Input without a valid marketId" }
```

Comment:
My preference is not using `requireNotNull` in production.  A deeplink to this view without marketId would trigger a crash here.

Reply:
it actually increases complexity quite a bit here to handle a null market id. i think it is reasonable to expect that this screen needs a market id set.

Reply:
i reverted this, but generally i think we should re-evaluate our thinking here.

i think we're over-indexing on theoretical crashes that would only happen if something is misconfigured. allowing this value to be nullable adds complexity and makes this code harder to read and maintain in the future.